Derive macro for `kani::Arbitrary` does not work on enums with a single variant.

<!--
If this is a security issue, please report it following the
[security reporting procedure](https://github.com/model-checking/kani/security/policy).
-->

<!--
Thank you for filing a bug report! 🐛
Please provide a short summary of the issue, along with the information necessary to replicate.
-->

I tried this code:
<!--
If the reproducing code is small, please post it here.
Otherwise, please attach the relevant files to this issue.
Alternatively, post a link to the repository and branch that exposes the issue.
-->
```rust
#[cfg_attr(kani, derive(kani::Arbitrary))]
enum Example {
    Variant,
}
```

using the following command line invocation:

```
kani example.rs
```

with Kani version: 0.56.0

I expected to see this happen: the file compiles successfully.

<!--
If Kani crashed, please include a backtrace in the code block by setting
`RUST_BACKTRACE=1` in your environment. E.g. `RUST_BACKTRACE=1 kani ...`
-->
Instead, the derived implementation of `kani::Arbitrary` does not compile successfully:
```
Kani Rust Verifier 0.56.0 (standalone)
error[E0283]: type annotations needed
 --> example.rs:1:25
  |
1 | #[cfg_attr(kani, derive(kani::Arbitrary))]
  |                         ^^^^^^^^^^^^^^^ cannot infer type
  |
  = note: cannot satisfy `_: kani::Arbitrary`
  = help: the following types implement trait `kani::Arbitrary`:
            ()
            (A, B)
            (A, B, C)
            (A, B, C, D)
            (A, B, C, D, E)
            (A, B, C, D, E, F)
            (A, B, C, D, E, F, G)
            (A, B, C, D, E, F, G, H)
          and 45 others
note: required by a bound in `kani::any`
 --> /Users/runner/work/kani/kani/library/kani/src/lib.rs:55:1
  = note: this error originates in the derive macro `kani::Arbitrary` which comes from the expansion of the macro `kani_core::kani_lib` (in Nightly builds, run with -Z macro-backtrace for more info)

```

--- a/library/kani_macros/src/derive.rs
+++ b/library/kani_macros/src/derive.rs
@@ -364,6 +364,12 @@ fn fn_any_enum(ident: &Ident, data: &DataEnum) -> TokenStream {
         quote! {
             panic!(#msg)
         }
+    } else if data.variants.len() == 1 {
+        let variant = data.variants.first().unwrap();
+        let init = init_symbolic_item(&variant.ident, &variant.fields);
+        quote! {
+            #ident::#init
+        }
     } else {
         let arms = data.variants.iter().enumerate().map(|(idx, variant)| {
             let init = init_symbolic_item(&variant.ident, &variant.fields);

--- a/tests/ui/derive-arbitrary/single_variant_enum/expected
+++ b/tests/ui/derive-arbitrary/single_variant_enum/expected
@@ -1,0 +1,16 @@
+Checking harness check_with_discriminant...
+SUCCESS\
+"assertion failed: disc == 42"
+
+Checking harness check_with_named_args...
+SUCCESS\
+"assertion failed: flag as u8 == 0 || flag as u8 == 1"
+2 of 2 cover properties satisfied
+
+Checking harness check_with_args...
+SUCCESS\
+"assertion failed: c <= char::MAX"
+2 of 2 cover properties satisfied
+
+Checking harness check_simple...
+1 of 1 cover properties satisfied

--- a/tests/ui/derive-arbitrary/single_variant_enum/single_variant_enum.rs
+++ b/tests/ui/derive-arbitrary/single_variant_enum/single_variant_enum.rs
@@ -1,0 +1,65 @@
+// Copyright Kani Contributors
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+//! Check that Kani can automatically derive Arbitrary enums.
+//! An arbitrary enum should always generate a valid arbitrary variant.
+
+extern crate kani;
+use kani::cover;
+
+#[derive(kani::Arbitrary)]
+enum Simple {
+    Empty,
+}
+
+#[kani::proof]
+fn check_simple() {
+    match kani::any::<Simple>() {
+        Simple::Empty => cover!(),
+    }
+}
+
+#[derive(kani::Arbitrary)]
+enum WithArgs {
+    Args(char),
+}
+
+#[kani::proof]
+fn check_with_args() {
+    match kani::any::<WithArgs>() {
+        WithArgs::Args(c) => {
+            assert!(c <= char::MAX);
+            cover!(c == 'a');
+            cover!(c == '1');
+        }
+    }
+}
+
+#[derive(kani::Arbitrary)]
+enum WithNamedArgs {
+    Args { flag: bool },
+}
+
+#[kani::proof]
+fn check_with_named_args() {
+    match kani::any::<WithNamedArgs>() {
+        WithNamedArgs::Args { flag } => {
+            cover!(flag as u8 == 0);
+            cover!(flag as u8 == 1);
+            assert!(flag as u8 == 0 || flag as u8 == 1);
+        }
+    }
+}
+
+#[derive(kani::Arbitrary)]
+enum WithDiscriminant {
+    Disc = 42,
+}
+
+#[kani::proof]
+fn check_with_discriminant() {
+    let v = kani::any::<WithDiscriminant>();
+    let disc = v as i8;
+    match v {
+        WithDiscriminant::Disc => assert!(disc == 42),
+    }
+}

